hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/ProtobufUtil.java

```diff
@@ -1072,6 +1077,22 @@ public static ClientProtos.Scan toScan(final Scan scan) throws IOException {
     return scanBuilder.build();
   }
 
+  public static List<Operation> toMulti(MultiRequest multiRequest) {
```

Comment:
I'm planning to add a test for this to `TestProtobufUtil`, and am very open to advice on how this implementation could/should be written differently

Reply:
this probably should throw an exception, since that seems more standard than catching it. Seems like you handle catching exceptions in your catchAll method anyway

Reply:
Otherwise, from a completeness perspective it's missing `hasServiceCall()` handling. There is no toServiceCall method yet, but you can just reverse the logic from RequestConverter.buildNoDataRegionActions which converts a RegionCoprocessorServiceExec to a CoprocessorServiceCall.

Reply:
> it's missing hasServiceCall() handling

We took this conversation offline and decided not to support service calls here, primarily because the `RegionCoprocessServiceExec` class [is not meant to be instantiated](https://github.com/apache/hbase/blob/9573466112712ea5823ddc48ebb3a084ff3e60e4/hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionCoprocessorServiceExec.java#L32-L33)